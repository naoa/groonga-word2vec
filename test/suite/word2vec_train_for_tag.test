register word2vec/word2vec

table_create Entries TABLE_NO_KEY
column_create Entries title COLUMN_SCALAR ShortText
column_create Entries content COLUMN_SCALAR ShortText

table_create Tokens TABLE_PAT_KEY ShortText --default_tokenizer TokenBigram
column_create Tokens entries_title COLUMN_INDEX|WITH_POSITION Entries title
column_create Tokens entries_content COLUMN_INDEX|WITH_POSITION Entries content

column_create Entries tags COLUMN_VECTOR Tokens
column_create Entries tags2 COLUMN_VECTOR Tokens

load --table Entries
[
{"title": "hoge", "content": "groonga + MySQL = mroonga.", "tags": ["Groonga", "Mroonga"], "tags2": ["hoge", "groonga"]},
{"title": "huge", "content": "rroonga", "tags": ["Groonga Database", "Rroonga"]},
{"title": "ato", "content": "rroonga", "tags": ["Groonga Database", "Rroonga"]},
{"title": "ago", "content": "rroonga", "tags": ["Groonga Database", "Rroonga"]},
{"title": "ago", "content": "rroonga", "tags": ["Groonga Database", "Rroonga"]},
{"title": "hugo", "content": "rroonga", "tags": ["Groonga Database", "Rroonga"]},
{"title": "ago", "content": "rroonga", "tags": ["Groonga Database", "Rroonga"]},
{"title": "ago", "content": "rroonga", "tags": ["Groonga Database", "Rroonga"]},
{"title": "ago", "content": "rroonga", "tags": ["Groonga Database", "Rroonga"]},
{"title": "age", "content": "rroonga", "tags": ["Groonga Database", "Rroonga"]}
]

word2vec_train Entries tags_,title,content,tags2_ --input_add_prefix "tag:" --no_mecab true

word2vec_distance "tag:Groonga_Database"
