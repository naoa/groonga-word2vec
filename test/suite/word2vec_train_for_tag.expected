register word2vec/word2vec
[[0,0.0,0.0],true]
table_create Entries TABLE_NO_KEY
[[0,0.0,0.0],true]
column_create Entries title COLUMN_SCALAR ShortText
[[0,0.0,0.0],true]
column_create Entries content COLUMN_SCALAR ShortText
[[0,0.0,0.0],true]
table_create Tokens TABLE_PAT_KEY ShortText --default_tokenizer TokenBigram
[[0,0.0,0.0],true]
column_create Tokens entries_title COLUMN_INDEX|WITH_POSITION Entries title
[[0,0.0,0.0],true]
column_create Tokens entries_content COLUMN_INDEX|WITH_POSITION Entries content
[[0,0.0,0.0],true]
column_create Entries tags COLUMN_VECTOR Tokens
[[0,0.0,0.0],true]
column_create Entries tags2 COLUMN_VECTOR Tokens
[[0,0.0,0.0],true]
load --table Entries
[
{"title": "hoge", "content": "groonga + MySQL = mroonga.", "tags": ["Groonga", "Mroonga"], "tags2": ["hoge", "groonga"]},
{"title": "huge", "content": "rroonga", "tags": ["Groonga Database", "Rroonga"]},
{"title": "ato", "content": "rroonga", "tags": ["Groonga Database", "Rroonga"]},
{"title": "ago", "content": "rroonga", "tags": ["Groonga Database", "Rroonga"]},
{"title": "ago", "content": "rroonga", "tags": ["Groonga Database", "Rroonga"]},
{"title": "hugo", "content": "rroonga", "tags": ["Groonga Database", "Rroonga"]},
{"title": "ago", "content": "rroonga", "tags": ["Groonga Database", "Rroonga"]},
{"title": "ago", "content": "rroonga", "tags": ["Groonga Database", "Rroonga"]},
{"title": "ago", "content": "rroonga", "tags": ["Groonga Database", "Rroonga"]},
{"title": "age", "content": "rroonga", "tags": ["Groonga Database", "Rroonga"]}
]
[[0,0.0,0.0],10]
word2vec_train Entries tags_,title,content,tags2_ --input_add_prefix "tag:" --no_mecab true
word2vec_distance "tag:Groonga_Database"
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      "tag:groonga_database",
      1.0
    ],
    [
      [
        "tag:rroonga",
        0.0745063871145248
      ],
      [
        "rroonga",
        0.057763833552599
      ],
      [
        "ago",
        0.0309041459113359
      ],
      [
        "</s>",
        0.0168687347322702
      ]
    ]
  ]
]
